addons/mail/static/src/chatter/web/chatter_patch.js

```diff
@@ -221,6 +250,16 @@ patch(Chatter.prototype, {
         }
     },
 
+    async onClickAttachFile(ev) {
+        if (this.state.thread.id) {
+            return;
+        }
+        const saved = await this.props.saveRecord?.();
+        if (!saved) {
+            return false;
+        }
```

Comment:
I discussed we should use `preventDefault` if possible instead of returning `false`.

Reply:
I tried to use `ev.stopPropagation()` and `ev.preventDefault()` instead of false to prevent opening attachment popup on non saved records and it doesn't work. It's because of the way fileUploader component capture the click (`onSelectFileButtonClick`).